// @mui/docs/legacy/Link/Link.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import * as React from 'react';
import clsx from 'clsx';
import { useRouter } from 'next/router';
import NextLink from 'next/link';
import MuiLink from '@mui/material/Link';
import { styled } from '@mui/material/styles';
import { useUserLanguage } from '../i18n';
import { useDocsConfig } from '../DocsProvider';

/**
 * File to keep in sync with:
 *
 * - /docs/src/modules/components/Link.tsx
 * - /examples/material-ui-nextjs-pages-router/src/Link.js
 * - /examples/material-ui-nextjs-pages-router-ts/src/Link.tsx
 */

// Add support for the sx prop for consistency with the other branches.
import { jsx as _jsx } from "react/jsx-runtime";
var Anchor = styled('a')({});
var NextLinkComposed = /*#__PURE__*/React.forwardRef(function NextLinkComposed(props, ref) {
  var to = props.to,
    linkAs = props.linkAs,
    replace = props.replace,
    scroll = props.scroll,
    shallow = props.shallow,
    prefetch = props.prefetch,
    _props$legacyBehavior = props.legacyBehavior,
    legacyBehavior = _props$legacyBehavior === void 0 ? true : _props$legacyBehavior,
    locale = props.locale,
    other = _objectWithoutProperties(props, ["to", "linkAs", "replace", "scroll", "shallow", "prefetch", "legacyBehavior", "locale"]);
  return /*#__PURE__*/_jsx(NextLink, {
    href: to,
    prefetch: prefetch,
    as: linkAs,
    replace: replace,
    scroll: scroll,
    shallow: shallow,
    passHref: true,
    locale: locale,
    legacyBehavior: legacyBehavior,
    children: /*#__PURE__*/_jsx(Anchor, _extends({
      "data-no-markdown-link": "true",
      ref: ref
    }, other))
  });
});
// A styled version of the Next.js Pages Router Link component:
// https://nextjs.org/docs/pages/api-reference/components/link
export var Link = /*#__PURE__*/React.forwardRef(function Link(props, ref) {
  var _props$activeClassNam = props.activeClassName,
    activeClassName = _props$activeClassNam === void 0 ? 'active' : _props$activeClassNam,
    as = props.as,
    classNameProps = props.className,
    href = props.href,
    legacyBehavior = props.legacyBehavior,
    linkAsProp = props.linkAs,
    locale = props.locale,
    noLinkStyle = props.noLinkStyle,
    prefetch = props.prefetch,
    replace = props.replace,
    role = props.role,
    scroll = props.scroll,
    shallow = props.shallow,
    other = _objectWithoutProperties(props, ["activeClassName", "as", "className", "href", "legacyBehavior", "linkAs", "locale", "noLinkStyle", "prefetch", "replace", "role", "scroll", "shallow"]);
  var router = useRouter();
  var pathname = typeof href === 'string' ? href : href == null ? void 0 : href.pathname;
  var routerPathname = router.pathname.replace('/[docsTab]', '');
  var shouldBeActive = routerPathname === pathname;
  var className = clsx(classNameProps, shouldBeActive && activeClassName);
  var userLanguage = useUserLanguage();
  var _useDocsConfig = useDocsConfig(),
    LANGUAGES_IGNORE_PAGES = _useDocsConfig.LANGUAGES_IGNORE_PAGES;
  var linkAs = linkAsProp || as || href;
  if (userLanguage !== 'en' && pathname && pathname.indexOf('/') === 0 && !LANGUAGES_IGNORE_PAGES(pathname) && !pathname.startsWith("/".concat(userLanguage, "/"))) {
    linkAs = "/".concat(userLanguage).concat(linkAs);
  }
  var nextjsProps = {
    to: href,
    linkAs: linkAs,
    replace: replace,
    scroll: scroll,
    shallow: shallow,
    legacyBehavior: legacyBehavior,
    prefetch: prefetch,
    locale: locale
  };
  if (noLinkStyle) {
    return /*#__PURE__*/_jsx(NextLinkComposed, _extends({
      className: className,
      ref: ref
    }, nextjsProps, other));
  }
  return /*#__PURE__*/_jsx(MuiLink, _extends({
    component: NextLinkComposed,
    className: className,
    ref: ref
  }, nextjsProps, other));
});
